src/live-programming.rst

```diff
@@ -26,6 +26,64 @@ Interactive Debuggers
 Program Visualization
 ---------------------
 
+**Program visualization** is the use of various techniques to enhance the human understanding of computer programs from distinct aspects including the structure of the source code and run-time behavior.
+
+There are **program visualization** systems designed to aid significantly the practice of professional programming and software engineering as well as the learning of computer science by novice programmers. And it turns out that **program visualization** doesn't have to deal with toy programs, e.g., `Frappe - Visualizing Code Dependency Graphs <https://softvis.wordpress.com/2018/08/31/frappe-visualizing-code-dependency-graphs/>`_.
+
+.. container:: bib-item
+
+  .. bibliography:: live-programming.bib
+    :filter: key == "price1993principled"
+
+ Note that **program visualization** is different from **software visualization** in that the latter includes the visualization of a high-level description of algorithm or a piece of software which is in contrast to **program visualization** where actual implemented code is visualized.
```

Comment:
This note should be in the introduction of this section, rather than as the summary of this paper. What was this paper specifically about?

Reply:
This is really a long paper going over dozens of software visualization systems and try to categorize them along the dimension described in the paper. And it also mentions some historical stuff and limitation of software visualization as well. I think most part of this paper is actually overlapping some other papers in some way(e.g. this paper spends a lot of time describing dimensions, but in a more detailed way than `maletic2002task`).
Personally, I can't think of a good way to put it in the compendium. What's your opinion?

Reply:
Rename the top level section to "Software Visualization", cite this paper first, then have a sub-section for Program VIsualization